docs/content/doc/advanced/hacking-on-gitea.en-us.md

```diff
@@ -15,41 +15,258 @@ menu:
 
 # Hacking on Gitea
 
-Familiarity with the existing [installation instructions](https://golang.org/doc/install)
-is required for this section.
+## Installing go and setting the GOPATH
 
-To contribute to Gitea, fork the project and work on the `master` branch.
+You should [install go](https://golang.org/doc/install) and set up your go
+environment correctly. In particular, it is recommended to set the `$GOPATH`
+environment variable and to add the go bin directory or directories
+`${GOPATH//://bin:}/bin` to the `$PATH`. See the Go wiki entry for
+[GOPATH](https://github.com/golang/go/wiki/GOPATH).
 
-Some internal packages are referenced using their respective Github URL. This can
-become problematic. To "trick" the Go tool into thinking this is a clone from the
-official repository, download the source code using "go get" and follow these instructions.
+**Note**: Whilst building, Gitea will download and build several packages
```

Comment:
Isn't this true only for Go 1.11 or stuff like `make lint`?

Reply:
so `make lint` is actually a thin wrapper around `revive` and `go` will get the tool as necessary. 

Or did you mean something else? I guess I've not made it clear that you will need to have make installed?

Reply:
I don't mean make, I think this implies when you run `go build` (either directly or through the makefile`, Gitea will download all go dependencies automatically. This is not the case, only for things like `make lint`/`make misspell-check`/etc.

Reply:
What wording would you suggest?

Reply:
Something like `when executing tasks which require external tools, like make misspell-check, Gitea will automatically download and build these tools. To be able for them to work properly, you need to add the go bin (usually $GOPATH/bin) to your path.`

And we should probably put a note about the facte that we're keeping all dependencies in the `vendor` folder while we're at it.

Reply:
Hmm. I don't think I know enough about `dep ensure` and the like to go there. Perhaps someone could do a follow up PR or suggest a section to put in wholesale about this.